From: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
Date: Wed, 16 May 2018 17:49:01 -0700
Subject: net/mlx5: Eswitch, Use 'kvfree()' for memory allocated by
 'kvzalloc()'
Patch-mainline: v4.18-rc1
Git-commit: e574978ae52a3fb1346bc37de2f2221dd878c065
References: bsc#1015342 FATE#321688 bsc#1015343 FATE#321689

When 'kvzalloc()' is used to allocate memory, 'kvfree()' must be used to
free it.

Fixes: fed9ce22bf8ae ("net/mlx5: E-Switch, Add API to create vport rx rules")
Signed-off-by: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/ethernet/mellanox/mlx5/core/eswitch_offloads.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/mellanox/mlx5/core/eswitch_offloads.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/eswitch_offloads.c
@@ -603,7 +603,7 @@ static int esw_create_vport_rx_group(str
 
 	esw->offloads.vport_rx_group = g;
 out:
-	kfree(flow_group_in);
+	kvfree(flow_group_in);
 	return err;
 }
 
